x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowTasksExecutor.java

```diff
@@ -185,7 +185,7 @@ protected void nodeOperation(final AllocatedPersistentTask task, final ShardFoll
                 return;
             }
 
-            if (ShardFollowNodeTask.shouldRetry(e)) {
+            if (ShardFollowNodeTask.shouldRetry(e) || e instanceof IndexNotFoundException) {
```

Comment:
Oh, actually, one comment; can you leave a comment here about this, since it's slightly different retry logic from elsewhere.